src/registry.cc

```diff
@@ -147,20 +147,28 @@ FloxFlakeInput::getSubtrees()
         }
       else
         {
-          auto root = this->getFlake()->openEvalCache()->getRoot();
-          if ( root->maybeGetAttr( "catalog" ) != nullptr )
+          try
             {
-              this->enabledSubtrees = std::vector<Subtree> { ST_CATALOG };
+              auto root = this->getFlake()->openEvalCache()->getRoot();
+              if ( root->maybeGetAttr( "catalog" ) != nullptr )
+                {
+                  this->enabledSubtrees = std::vector<Subtree> { ST_CATALOG };
+                }
+              else if ( root->maybeGetAttr( "packages" ) != nullptr )
+                {
+                  this->enabledSubtrees = std::vector<Subtree> { ST_PACKAGES };
+                }
+              else if ( root->maybeGetAttr( "legacyPackages" ) != nullptr )
+                {
+                  this->enabledSubtrees = std::vector<Subtree> { ST_LEGACY };
+                }
+              else { this->enabledSubtrees = std::vector<Subtree> {}; }
             }
-          else if ( root->maybeGetAttr( "packages" ) != nullptr )
+          catch ( const nix::EvalError &err )
```

Comment:
@aakropotkin I was just guessing this is the right error to catch here, do you know if that's correct?

Reply:
Yeah you got the right one.
This is the underlying `throw` : https://github.com/flox/pkgdb/blob/main/src/pkgdb/write.cc#L446